vault/src/issue.rs

```diff
@@ -164,39 +164,65 @@ async fn process_transaction_and_execute_issue<B: BitcoinCoreApi + Send + Sync +
 ) -> Result<(), Error> {
     let addresses = transaction.extract_output_addresses::<BtcAddress>();
     let mut issue_requests = issue_set.0.lock().await;
-    if let Some(address) = addresses
-        .iter()
-        .find(|&vout| issue_requests.contains_value(vout))
-    {
+    if let Some((issue_id, address)) = addresses.iter().find_map(|address| {
+        let issue_id = issue_requests.get_key_for_value(address)?;
+        Some((issue_id.clone(), address.clone()))
+    }) {
+        let issue = provider.get_issue_request(issue_id).await?;
         // tx has output to address
-        if let Some(issue_id) = issue_requests.remove_value(address) {
-            info!("Found tx for issue with id {}", issue_id);
-
-            // at this point we know that the transaction has `num_confirmations` on the bitcoin chain,
-            // but the relay can introduce a delay, so wait until the relay also confirms the transaction.
-            provider
-                .wait_for_block_in_relay(
-                    H256Le::from_bytes_le(&block_hash.to_vec()),
-                    num_confirmations,
-                )
-                .await?;
-
-            // found tx, submit proof
-            let txid = transaction.txid();
-            let raw_tx = btc_rpc.get_raw_tx_for(&txid, &block_hash).await?;
-            let proof = btc_rpc.get_proof_for(txid.clone(), &block_hash).await?;
-
-            info!("Executing issue with id {}", issue_id);
-
-            // this will error if someone else executes the issue first
-            provider
-                .execute_issue(
-                    issue_id,
-                    H256Le::from_bytes_le(&txid.as_hash()),
-                    proof,
-                    raw_tx,
-                )
-                .await?;
+        match transaction.get_payment_amount_to(address) {
+            None => {
+                // this should never happen, so use WARN
+                warn!(
+                    "Could not extract payment amount for transaction {}",
+                    transaction.txid()
+                );
+                return Ok(());
+            }
+            Some(transferred) => {
+                let transferred = transferred as u128;
+                if transferred == issue.amount {
+                    info!("Found tx for issue with id {}", issue_id);
+                } else {
+                    info!(
+                        "Found tx for issue with id {}. Expected amount = {}, got {}",
+                        issue_id, issue.amount, transferred
+                    );
+                }
+
+                if transferred < issue.amount {
```

Comment:
Happy to have this now, but think it may change once we allow marginal issuance on the parachain.

Reply:
The partial issuance will require some thought on how to deal with griefing, and how to show it in the UI. Until then it is good to have this, since a lot of users are worried by the InsufficientValue error